File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Go/s012761786_commented.go
package main

import (
	"fmt"
)

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func main() {
	var n, a, b int64
	fmt.Scan(&n, &a, &b)

	sa := a - b
	if sa < 0 {
		sa = -sa
	}

	var ans int64

	if sa%2 == 0 {
		ans = sa / 2
	} else {
		AAA := min(n-a, a-1)
		BBB := min(n-b, b-1)
		ans = min(AAA, BBB)

		sa -= 
		ans += 1
		ans += sa / 2
	}

	fmt.Println(ans)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/C++/s012761786_commented.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
using namespace std;

int main() {
    long n, a, b;
    cin >> n >> a >> b;
    long sa = labs(a - b);
    long ans = 0;
    if (sa % 2 == 0) {
        ans = sa / 2;
    } else {
        long AAA = min((n - a), (a - 1));
        long BBB = min((n - b), (b - 1));
        ans = min(AAA, BBB);
        sa -= 1;
        ans += 1;
        ans += sa / 2;
    }
    cout << ans << endl;
    return 0;
}
// End of Code
